dds/DCPS/DCPS_Utils.cpp

```diff
@@ -464,21 +477,35 @@ bool repr_to_encoding_kind(DDS::DataRepresentationId_t repr, Encoding::Kind& kin
   case DDS::XCDR2_DATA_REPRESENTATION:
     kind = Encoding::KIND_XCDR2;
     break;
+  case DDS::UNALIGNED_CDR_DATA_REPRESENTATION:
+    kind = Encoding::KIND_UNALIGNED_CDR;
+    break;
   default:
     return false;
   }
   return true;
 }
 
-DDS::DataRepresentationIdSeq get_effective_data_rep_qos(const DDS::DataRepresentationIdSeq& qos, bool reader)
+DDS::DataRepresentationIdSeq get_writer_effective_data_rep_qos(const DDS::DataRepresentationIdSeq& qos,
+  bool cdr_encapsulated)
 {
   if (qos.length() == 0) {
     DDS::DataRepresentationIdSeq ids;
-    ids.length(reader ? 2 : 1);
-    ids[0] = DDS::XCDR2_DATA_REPRESENTATION;
-    if (reader) {
-      ids[1] = DDS::XCDR_DATA_REPRESENTATION;
-    }
+    ids.length(1);
+    ids[0] = cdr_encapsulated ? DDS::XCDR2_DATA_REPRESENTATION : DDS::UNALIGNED_CDR_DATA_REPRESENTATION;
+    return ids;
+  }
+  return qos;
+}
+
+DDS::DataRepresentationIdSeq get_reader_effective_data_rep_qos(const DDS::DataRepresentationIdSeq& qos)
+{
+  if (qos.length() == 0) {
+    DDS::DataRepresentationIdSeq ids;
+    ids.length(3);
+    ids[0] = DDS::XCDR_DATA_REPRESENTATION;
+    ids[1] = DDS::XCDR2_DATA_REPRESENTATION;
+    ids[2] =  DDS::UNALIGNED_CDR_DATA_REPRESENTATION;
```

Comment:
Doesn't it need to be in the DDS namespace to be defined alongside the others or are you saying to define it seperately?

Reply:
It doesn't need to be in any specific namespace, these are just integer constants (not an enum).  It should not be in the DDS namespace because it's not in the spec.